pkg/webhook/admission/defaulter_test.go

```diff
@@ -66,3 +72,124 @@ func (d *TestDefaulter) Default() {
 		d.Replica = 2
 	}
 }
+
+/*
+*
+BEFORE (without isByteArrayEqual)
+goos: darwin
+goarch: arm64
+pkg: sigs.k8s.io/controller-runtime/pkg/webhook/admission
+BenchmarkHandle
+BenchmarkHandle/benchmark_handle_function
+BenchmarkHandle/benchmark_handle_function-10         	   22878	     50394 ns/op
+
+AFTER (with isByteArrayEqual)
+goos: darwin
+goarch: arm64
+pkg: sigs.k8s.io/controller-runtime/pkg/webhook/admission
+BenchmarkHandle
+BenchmarkHandle/benchmark_handle_function
+BenchmarkHandle/benchmark_handle_function-10         	   44620	     26178 ns/op
+*/
+func BenchmarkHandle(b *testing.B) {
```

Comment:
These two benchmarks are not representative, as you are only benchmarking the "objects are equal case" and not the "objects are not equal", in which case your change worsens performance

Reply:
Thanks for your prompt reply @alvaroaleman 
I've add more benchmark tests regarding equal case and not equal case.
```
BenchmarkPatchResponseFromRawAndIsByteArrayEqual/benchmark_PatchResponseFromRaw_function-10         	   52774	     22749 ns/op
BenchmarkPatchResponseFromRawAndIsByteArrayEqual/benchmark_isByteArrayEqual_function-10             	 5252599	       230.7 ns/op
```
To add more details to this:
1) By benchmarking, comparing two byte arrays takes far less time than the recursive reflect.DeepEqual, which is about 1/100 according to my above benchmark
2) Most of the Apiserver's patch/update requests will still get same object after going through all mutating webhooks. as mutating webhooks most of the time take effect at the beginning of a Pod's lifecycle (or other resources lifecycle)
3) In our production cluster, this change "reducing recursive reflect.DeepEqual" can save more than 40% cpu time for webhook.